### PR TITLE
Bug fixes for affine-mapper

### PR DESCRIPTION
**Pull request by max-collins**
## Linked Issue
Fixes # (issue number)

## Implementation Checklist
- [x] My class inherits from `PolynomialBasis` (if applicable).
- [x] I have implemented all abstract methods (`evaluate_basis`, `fit`, etc.).
- [x] I have used `numpy` vectorization to avoid inefficient Python loops.
- [x] Documentation strings (docstrings) are provided for all public methods.

### 1. Mathematical Integrity
- [ ] **Convergence:** Does the implementation maintain expected accuracy? (Checked for Runge's phenomenon or stability).
- [ ] **Edge Cases:** Does it handle single points or non-equally spaced nodes correctly?
- [ ] **Domain Mapping:** Does the class correctly map the input domain to the basis domain (e.g., [-1, 1] for Chebyshev)?

### 2. API & Style
- [ ] **Method Signatures:** Methods match the Abstract Base Class exactly.
- [ ] **Naming:** Follows `CamelCase` for classes and `snake_case` for functions/variables.
- [ ] **Clean Code:** No "magic numbers"; constants are named and explained.

### 3. Testing Quality
- [ ] **Coverage:** Unit tests cover typical usage and edge cases.
- [ ] **Precision:** Tests use `np.testing.assert_allclose` instead of `==`.
- [ ] **Validation:** There is a test proving the basis can exactly reconstruct a polynomial of its own degree.

## Screenshots / Plots

### CODE REVIEW

**Comment by mtezzele** [comment]
Please use more descriptive commits and provide a description of your PR. What was the problem? How did you fix it?
Also, please try to rebase this branch.

**Comment by max-collins** [comment]
The fit method was pulling back the y_values to the reference interval which was incorrect. The method has been updated to not adjust the y_values.

File: src/numanalysislib/basis/affine.py [hunk 1]
```diff
@@ -47,21 +44,26 @@ def evaluate_basis(self, index: int, x: np.ndarray) -> np.ndarray:
...
+        return self.basis.fit(x_nodes_hat, y_nodes)
 
 
     def pull_back(self, x: float) -> float:
         """
         Maps $x \in [a, b] \to \hat{x} \in [\hat{a}, \hat{b}]
 
-        Inputs:
+        Args:
         x: element of physical interval to be mapped into reference interval
```
Comment (mtezzele) [inline src/numanalysislib/basis/affine.py:64]: I am pretty sure this will not be rendered correctly by the doc (follow the updated instructions in the readme on how to generate it locally).
